### PR TITLE
sql: make example in RBAC doc easier to understand

### LINKED ISSUE #2496: RBAC 文档描述不易操作，缺少场景说明
## Feature Request

**This repository is ONLY used to solve problems related to DOCS-CN.
For other issues (TiDB, TiKV, etc), please move to [other repositories](https://github.com/pingcap/).**

1. Did you encounter a certain issue about the documentation? If yes, please describe the issue that relates to your feature request.

[RBAC 文档](https://pingcap.com/docs-cn/dev/reference/security/role-based-access-control/)中的例子操作起来不连贯，用户难以理解使用场景。

2. Please describe your suggestion or addition.

1. 补充更容易操作和理解的例子
2.对使用场景进行解释

3. If convenient, please provide reference materials (Blog, Stack Overflow, etc).

### PR DESCRIPTION
**Pull request by imtbkcat**
### What is changed, added or deleted? (Required)

To close #2496 
make example easier to understand.
user can try RBAC by simply copy and paste sql in doc. 

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
Please involve a technical review. @imtbkcat

File: reference/security/role-based-access-control.md [hunk 1]
```diff
@@ -15,72 +15,69 @@ TiDB 的基于角色的访问控制 (RBAC) 系统的实现类似于 MySQL 8.0 
...
+为角色授予权限和为用户授予权限操作相同，可参考 [TiDB 权限管理](/reference/security/privilege-system.md)。
 
-删除角色 r_1 和 r_2：
+为 `app_reader` 角色授予数据库 `app_db` 的读权限：
 
 {{< copyable "sql" >}}
 
 ```sql
-DROP ROLE `r_1`@`%`, `r_2`@`%`;
+GRANT SELECT ON app_db.* TO 'app_reader'@'%';
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
GRANT SELECT ON app_db.* TO 'app_read'@'%';
```

File: reference/security/role-based-access-control.md [hunk 2]
```diff
@@ -15,72 +15,69 @@ TiDB 的基于角色的访问控制 (RBAC) 系统的实现类似于 MySQL 8.0 
...
 角色会被保存在 `mysql.user` 表中，如果表中有同名角色或用户，角色会创建失败并报错。
 创建角色的用户需要拥有 `CREATE ROLE` 或 `CREATE USER` 权限。
 
-### 删除角色
+### 授予角色权限
+
+为角色授予权限和为用户授予权限操作相同，可参考 [TiDB 权限管理](/reference/security/privilege-system.md)。
 
-删除角色 r_1 和 r_2：
+为 `app_reader` 角色授予数据库 `app_db` 的读权限：
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
为 `app_read` 角色授予数据库 `app_db` 的读权限：
```
前面说创建的角色时"app_read"和"app_write"？这里是写错了？

File: reference/security/role-based-access-control.md [hunk 3]
```diff
@@ -15,72 +15,69 @@ TiDB 的基于角色的访问控制 (RBAC) 系统的实现类似于 MySQL 8.0 
...
 
-这个操作会清除角色在 `mysql.user` 表里面的记录项，并且清除在授权表里面的相关记录，解除和其相关的授权关系。
-执行删除角色的用户需要拥有 `DROP ROLE` 或 `DROP USER` 权限。
-
-### 授予角色权限
-
-为角色授予权限和为用户授予权限操作相同，可参考 [TiDB 权限管理](/reference/security/privilege-system.md)。
-
-为 `analyst` 角色授予数据库 `test` 的读权限：
+为 `app_writer` 角色授予数据库 `app_db` 的写权限：
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
为 `app_write` 角色授予数据库 `app_db` 的写权限：
```
ditto

File: reference/security/role-based-access-control.md [hunk 4]
```diff
@@ -15,72 +15,69 @@ TiDB 的基于角色的访问控制 (RBAC) 系统的实现类似于 MySQL 8.0 
...
 
 {{< copyable "sql" >}}
 
 ```sql
-GRANT SELECT ON test.* TO 'analyst'@'%';
+GRANT INSERT, UPDATE, DELETE ON app_db.* TO 'app_writer'@'%';
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
GRANT INSERT, UPDATE, DELETE ON app_db.* TO 'app_write'@'%';
```

File: reference/security/role-based-access-control.md [hunk 5]
```diff
@@ -15,72 +15,69 @@ TiDB 的基于角色的访问控制 (RBAC) 系统的实现类似于 MySQL 8.0 
...
 ```sql
-GRANT ALL PRIVILEGES ON *.* TO 'analyst'@'%';
+GRANT ALL ON app_db.* TO 'app_developer';
 ```
 
-### 收回权限
+### 将角色授予给用户
 
-`REVOKE` 语句与 `GRANT` 对应：
+假设有有一个用户拥有开发者角色，可以对 `app_db` 的所有操作权限，另外有两个用户拥有 `app_db` 的只读权限。还有一个用户拥有 `app_db` 的读写权限。
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
假设有一个用户拥有开发者角色，可以对 `app_db` 的所有操作权限；另外有两个用户拥有 `app_db` 的只读权限；还有一个用户拥有 `app_db` 的读写权限。
```

File: reference/security/role-based-access-control.md [hunk 6]
```diff
@@ -238,99 +290,71 @@ SELECT CURRENT_ROLE();
...
-| Grants for u1@localhost                     |
-+---------------------------------------------+
-| GRANT USAGE ON *.* TO `u1`@`localhost`      |
-| GRANT `r1`@`%`,`r2`@`%` TO `u1`@`localhost` |
-+---------------------------------------------+
+REVOKE 'app_read' FROM 'read_user1'@'localhost', 'read_user2'@'localhost';
 ```
 
-可以通过使用 `SHOW GRANTS` 的 `USING` 选项来查看角色对应的权限。
+解除角色 `app_read`, `app_write` 与用户 `rw_user1@localhost` 的授权关系。
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
解除角色 `app_read`、`app_write` 与用户 `rw_user1@localhost` 的授权关系。
```

File: reference/security/role-based-access-control.md [hunk 7]
```diff
@@ -238,99 +290,71 @@ SELECT CURRENT_ROLE();
...
-
-```
-+---------------------------------------------------------------------+
-| Grants for u1@localhost                                             |
-+---------------------------------------------------------------------+
-| GRANT USAGE ON *.* TO `u1`@`localhost`                              |
-| GRANT Select, Insert, Update, Delete ON `db1`.* TO `u1`@`localhost` |
-| GRANT `r1`@`%`,`r2`@`%` TO `u1`@`localhost`                         |
-+---------------------------------------------------------------------+
+DROP ROLE 'app_read', 'app_write';;
```
Comment (yikeke) [inline reference/security/role-based-access-control.md]:
```suggestion
DROP ROLE 'app_read', 'app_write';
```

**Comment by tiancaiamao** [comment]
LGTM

**Comment by Hangzhi** [comment]
@yikeke: I'd like to translate this PR

**Comment by yikeke** [comment]
> @yikeke: I'd like to translate this PR

Sure, this PR is assigned to you!

**Comment by yikeke** [comment]
> @yikeke: I'd like to translate this PR

Hi @Hangzhi  are you still translating this PR? If not, I'll assign another person to do it.

**Pull request merged**